lib/taskana-core/src/main/java/pro/taskana/task/internal/TaskServiceImpl.java

```diff
@@ -645,10 +637,14 @@ public void forceDeleteTask(String taskId)
     try {
       taskanaEngine.openConnection();
       // use only elements we are authorized for
-      Pair<List<String>, BulkOperationResults<String, TaskanaException>> resultsPair =
-          filterForAuthorizedTasks(taskIds);
+      Pair<List<MinimalTaskSummary>, BulkLog> resultsPair =
+          serviceLevelHandler.filterTasksForExistenceAndAuthorization(taskIds);
```

Comment:
Placing it into TaskServiceImpl creates dependency loops because then TaskServiceImpl depends on ServiceLevelHandler and ServiceLevelHandler depends on TaskServiceImpl.
If we want to move it out of ServiceLevelHandler, we should create a general TaskServiceHelper class that can be called from TaskServiceImpl, ServiceLevelHandler, etc. 
In this way we would create a loop-free dependency graph...